iroh-relay/src/server.rs

```diff
@@ -122,6 +138,33 @@ pub struct StunConfig {
     pub bind_addr: SocketAddr,
 }
 
+/// Configuration for the QUIC server.
+// TODO(ramfox): limits? Limiting connections or accepted # of connections from a single IP? from a single node ID?
+#[derive(Debug)]
+pub struct QuicConfig {
+    /// The socket address on which the QUIC server should bind.
+    ///
+    /// Normally you'd chose port `7842`, see [`crate::defaults::DEFAULT_QUIC_PORT`].
+    pub bind_addr: SocketAddr,
+    /// The TLS server configuration for the QUIC server.
+    #[debug(skip)]
+    pub server_config: QuicServerConfig,
+}
+
+impl QuicConfig {
+    /// Create a new [`QuicConfig`] from a [`TlsConfig`].
```

Comment:
ahh yes, thank you. this used to be MUCH more complicated, but as I understood what was going on better, it slimmed down to this! I'll see what removing it looks like 😂

Reply:
Okay, so now, with the `quic_bind_addr` in the TlsConfig, it actually makes sense to keep this and pass in a TlsConfig so that the QuicConfig is supplied with all the correct pieces.